### ISSUE TITLE
Full Name is not displayed

### ISSUE THREAD

**Issue by dkontod** [volkanceylan/Serenity#1783]
Hello.
I have used the Full Name implementation as it is described - and worked for MovieTutorial - in the guide but while it is displayed at the dialog title, it does not appear at the grid (serene 2.7.2):
AddressBookRow.cs:

//...........
            [DisplayName("Full Name"),
                Expression("(t0.LastName + ' ' + t0.FirstName)"), QuickSearch]
            public String Fullname
            {
                get { return Fields.FullName[this]; }
                set { Fields.FullName[this] = value; }
            }
            public partial class RowFields { public StringField FullName; }
//.......

AddressBookColumns,cs:

//....
        public Int32 AddressId { get; set; }
        public Int32 AddressType { get; set; }
        [Hidden]
        public String LastName { get; set; }
        [Hidden]
        public String FirstName { get; set; }
        [EditLink]
        public String FullName { get; set; }
//......

Any ideas?

**Comment by edson** [comment]
If you mean the column is not showing, see if it is hidden in the column picker dialog.
This happens when you enable userpreferences storage and add a new column and the preference is already saved.

Best Regards

**Comment by dkontod** [comment]
Thank you for your response edson.
No, I mean that the column is showing but empty. FullName is populated correctly, because I can see it at the dialog title: Edit Address (<FullName>)

**Comment by edson** [comment]
Oh, I see.

Well.. Check if FullName is returned filled in browser console, or if it has errors..
Or maybe you need to check if there's a NULL in either column, LastName and FirstName.

Your expression would have to be something like:

`Expression("( isNull(t0.LastName,'') + ' ' +  isNull(t0.FirstName,''))")`

Best Regards

**Comment by dkontod** [comment]
I see. No. Either FirstName and LastName are not null, since they are required to the table and the row.cs. Something else is happening having to do with the release update. You see the tutorial is at a previous serenity release and this issue is at the latest.

**Comment by Estrusco** [comment]
Add  [MinSelectLevel(SelectLevel.List)]

**Comment by dkontod** [comment]
Thank you Estrusco.
You mean at the row.cs or the columns.cs?

**Comment by dkontod** [comment]
Nope. Not working:
            [DisplayName("Full Name"),
                Expression("(t0.LastName + ' ' + t0.FirstName)"), QuickSearch]
            [MinSelectLevel(SelectLevel.List)]
            public String Fullname
            {
                get { return Fields.FullName[this]; }
                set { Fields.FullName[this] = value; }
            }

**Comment by volkanceylan** [comment]
@dkontod you really have a problem with reading. In tutorial it was clearly stated:

>In our MasterDetailRelation attribute, we specified an extra property, IncludeColumns:
[MasterDetailRelation(foreignKey: "MovieId", IncludeColumns = "PersonFullname")]
This ensures that PersonFullname field on cast list is selected on retrieve. Otherwise, it wouldn't be loaded as only table fields are selected by default. When you open a movie dialog with existing cast list, full name would be empty.

Please don't waste your and our time. I'm really tired of answering same questions. This question is asked a million times.

**Comment by volkanceylan** [comment]
These are not issues, these are your ignorance, see

https://github.com/volkanceylan/Serenity/issues/1610

**Comment by dkontod** [comment]
@volkanceylan  thank you for your answer. Yes, I have read this topic. And yes, I have ignorance. But no, this is not the case. There is no Master-Detail relationship here. There is no foreign key. There is only one table with FirstName and LastName fields. And I have made the exact implementation as in the tutorial (by my own I admit) where it was working fine at the Person grid. The only difference is that this project is at a newer version than the tutorial.
As I have already told you you have done an amazing job and you continue doing it. But please, don't insult all of us who are trying upon your creation.

**Comment by volkanceylan** [comment]
Employee name in OrdersGrid also comes from a similar CONCAT operation, so thats clearly not a version issue.

**Comment by dkontod** [comment]
This is the result I'm trying to achieve:

![image](https://cloud.githubusercontent.com/assets/24634785/21865469/de5a7f42-d84e-11e6-96fb-b5a76496e90e.png)

PersonColumns.cs:

    [ColumnsScript("MovieDB.Person")]
    [BasedOnRow(typeof(Entities.PersonRow))]
    public class PersonColumns
    {
        [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
        public Int32 PersonId { get; set; }
        [EditLink]
        public String Fullname { get; set; }
        [EditLink]
        public String Firstname { get; set; }
        [EditLink]
        public String Lastname { get; set; }
        [EditLink]
//...

PersonRow.cs:

    public sealed class PersonRow : Row, IIdRow, INameRow
    {
        [DisplayName("Person Id"), Identity]
        public Int32? PersonId
        {
            get { return Fields.PersonId[this]; }
            set { Fields.PersonId[this] = value; }
        }

        [DisplayName("Firstname"), Size(50), NotNull]
        public String Firstname
        {
            get { return Fields.Firstname[this]; }
            set { Fields.Firstname[this] = value; }
        }

        [DisplayName("Lastname"), Size(50), NotNull]
        public String Lastname
        {
            get { return Fields.Lastname[this]; }
            set { Fields.Lastname[this] = value; }
        }
        [DisplayName("Full Name"),
        Expression("(t0.Firstname + ' ' + t0.Lastname)"), QuickSearch]
        public String Fullname
        {
            get { return Fields.Fullname[this]; }
            set { Fields.Fullname[this] = value; }
        }
//...

PersonGrid.ts:

namespace MovieTutorial.MovieDB {
    
    @Serenity.Decorators.registerClass()
    export class PersonGrid extends Serenity.EntityGrid<PersonRow, any> {
        protected getColumnsKey() { return 'MovieDB.Person'; }
        protected getDialogType() { return PersonDialog; }
        protected getIdProperty() { return PersonRow.idProperty; }
        protected getLocalTextPrefix() { return PersonRow.localTextPrefix; }
        protected getService() { return PersonService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}

**Comment by dkontod** [comment]
And this is what I see in my project

![image](https://cloud.githubusercontent.com/assets/24634785/21865996/014d69f4-d851-11e6-9e9e-4014857c300e.png)

**Comment by volkanceylan** [comment]
Show full source for row and columns

**Comment by dkontod** [comment]
Thank you very much @volkanceylan for your time. The truth is that it is a project which I have to present tomorrow (It's 00:15 now) and I'm desperate.
`
//AddressBookRow.cs
namespace AddressDB.AddressBookDB.Entities
{
    using System.Collections.Generic;
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Data.Mapping;
    using System;
    using System.ComponentModel;
    using System.IO;

    [ConnectionKey("Default"), DisplayName("AddressBook"), InstanceName("AddressBook"), TwoLevelCached]
    [ReadPermission("AddressBookDB:AddressBook:Read")]
    [InsertPermission("AddressBookDB:AddressBook:Insert")]
    [UpdatePermission("AddressBookDB:AddressBook:Update")]
    [DeletePermission("AddressBookDB:AddressBook:Delete")]
    [LookupScript("AddressBookDB.AddressBookRow")]
    public sealed class AddressBookRow : Row, IIdRow, INameRow
    {        
            #region Address Id
            [DisplayName("Address Id"), Column("AddressID"), Identity]
            public Int32? AddressId { get { return Fields.AddressId[this]; } set { Fields.AddressId[this] = value; } }
            public partial class RowFields { public Int32Field AddressId; }
            #endregion AddressId
                
            #region Address Type
            [DisplayName("Address Type"), NotNull, ForeignKey("[adr].[AddressType]", "AddressTypeID"), LeftJoin("jAddressType"), TextualField("AddressTypeAddressTypeName")]
            [LookupEditor(typeof(AddressBookDB.Entities.AddressTypeRow), InplaceAdd = true)]
            public Int32? AddressType { get { return Fields.AddressType[this]; } set { Fields.AddressType[this] = value; } }
            public partial class RowFields { public Int32Field AddressType; }
            #endregion AddressType
                
            #region Last Name
            [DisplayName("Last Name"), Size(100), NotNull]
            public String LastName { get { return Fields.LastName[this]; } set { Fields.LastName[this] = value; } }
            public partial class RowFields { public StringField LastName; }
            #endregion LastName
                
            #region First Name
            [DisplayName("First Name"), Size(100), NotNull]
            public String FirstName { get { return Fields.FirstName[this]; } set { Fields.FirstName[this] = value; } }
            public partial class RowFields { public StringField FirstName; }
        #endregion FirstName
            
            #region FullName
            [DisplayName("Full Name"),
                Expression("(t0.LastName + ' ' + t0.FirstName)"), QuickSearch]
            [MinSelectLevel(SelectLevel.List)]
            public String Fullname
            {
                get { return Fields.FullName[this]; }
                set { Fields.FullName[this] = value; }
            }
            public partial class RowFields { public StringField FullName; }
            #endregion FullName

            #region Gender
            [DisplayName("Gender"), NotNull]
            public Gender? Gender { get { return (Gender?)Fields.Gender[this]; } set { Fields.Gender[this] = (Int32?)value; } }
            public partial class RowFields { public Int32Field Gender; }
            #endregion Gender
                
            #region Father Name
            [DisplayName("Father Name"), Size(100), NotNull]
            public String FatherName { get { return Fields.FatherName[this]; } set { Fields.FatherName[this] = value; } }
            public partial class RowFields { public StringField FatherName; }
            #endregion FatherName
                
            #region Mother Name
            [DisplayName("Mother Name"), Size(100)]
            public String MotherName { get { return Fields.MotherName[this]; } set { Fields.MotherName[this] = value; } }
            public partial class RowFields { public StringField MotherName; }
            #endregion MotherName
                
            #region Address
            [DisplayName("Address"), Size(300)]
            public String Address { get { return Fields.Address[this]; } set { Fields.Address[this] = value; } }
            public partial class RowFields { public StringField Address; }
            #endregion Address
                
            #region City
            [DisplayName("City"), Size(100)]
            public String City { get { return Fields.City[this]; } set { Fields.City[this] = value; } }
            public partial class RowFields { public StringField City; }
            #endregion City
                
            #region Post Code
            [DisplayName("Post Code"), Size(10)]
            public String PostCode { get { return Fields.PostCode[this]; } set { Fields.PostCode[this] = value; } }
            public partial class RowFields { public StringField PostCode; }
            #endregion PostCode
                
            #region Profession
            [DisplayName("Profession"), Size(100)]
            public String Profession { get { return Fields.Profession[this]; } set { Fields.Profession[this] = value; } }
            public partial class RowFields { public StringField Profession; }
            #endregion Profession

            #region Communication
        [DisplayName("Communication"), NotMapped]
        
        public List<CommunicationRow> Communication
        {
            get { return Fields.Communication[this]; }
            set { Fields.Communication[this] = value; }
        }
        public partial class RowFields { public RowListField<CommunicationRow> Communication; }
        #endregion Communication

            #region Birth Date
        [DisplayName("Birth Date")]
            public DateTime? BirthDate { get { return Fields.BirthDate[this]; } set { Fields.BirthDate[this] = value; } }
            public partial class RowFields { public DateTimeField BirthDate; }
            #endregion BirthDate
                
            #region Constituency
            [DisplayName("Constituency"), ForeignKey("[adr].[Constituency]", "ConstituencyID"), LeftJoin("jConstituency"), TextualField("ConstituencyConstituencyName")]
            [LookupEditor(typeof(AddressBookDB.Entities.ConstituencyRow), InplaceAdd = true)]
            public Int32? Constituency { get { return Fields.Constituency[this]; } set { Fields.Constituency[this] = value; } }
            public partial class RowFields { public Int32Field Constituency; }
            #endregion Constituency
                
            #region Voting Center
            [DisplayName("Voting Center"), ForeignKey("[adr].[VotingCenter]", "VotingCenterID"), LeftJoin("jVotingCenter"), TextualField("VotingCenterVotingCenterName")]
            [LookupEditor(typeof(AddressBookDB.Entities.VotingCenterRow), InplaceAdd = true)]
            public Int32? VotingCenter { get { return Fields.VotingCenter[this]; } set { Fields.VotingCenter[this] = value; } }
            public partial class RowFields { public Int32Field VotingCenter; }
            #endregion VotingCenter
                
            #region Polling Station
            [DisplayName("Polling Station"), Size(30)]
            public String PollingStation { get { return Fields.PollingStation[this]; } set { Fields.PollingStation[this] = value; } }
            public partial class RowFields { public StringField PollingStation; }
            #endregion PollingStation
                
            #region Description
            [DisplayName("Description"), Size(300)]
            public String Description { get { return Fields.Description[this]; } set { Fields.Description[this] = value; } }
            public partial class RowFields { public StringField Description; }
            #endregion Description
                
            #region Notes
            [DisplayName("Notes"), Size(1000)]
            public String Notes { get { return Fields.Notes[this]; } set { Fields.Notes[this] = value; } }
            public partial class RowFields { public StringField Notes; }
            #endregion Notes
                
            #region Photo
            [DisplayName("Photo"), Size(100)]
            public String Photo { get { return Fields.Photo[this]; } set { Fields.Photo[this] = value; } }
            public partial class RowFields { public StringField Photo; }
            #endregion Photo
                
            #region Number1
            [DisplayName("Number1")]
            public Int32? Number1 { get { return Fields.Number1[this]; } set { Fields.Number1[this] = value; } }
            public partial class RowFields { public Int32Field Number1; }
            #endregion Number1
                
            #region Number2
            [DisplayName("Number2")]
            public Int32? Number2 { get { return Fields.Number2[this]; } set { Fields.Number2[this] = value; } }
            public partial class RowFields { public Int32Field Number2; }
            #endregion Number2
                
            #region Number3
            [DisplayName("Number3")]
            public Int32? Number3 { get { return Fields.Number3[this]; } set { Fields.Number3[this] = value; } }
            public partial class RowFields { public Int32Field Number3; }
            #endregion Number3
                
            #region Number4
            [DisplayName("Number4")]
            public Int32? Number4 { get { return Fields.Number4[this]; } set { Fields.Number4[this] = value; } }
            public partial class RowFields { public Int32Field Number4; }
            #endregion Number4
                
            #region Number5
            [DisplayName("Number5")]
            public Int32? Number5 { get { return Fields.Number5[this]; } set { Fields.Number5[this] = value; } }
            public partial class RowFields { public Int32Field Number5; }
            #endregion Number5
                
            #region Date1
            [DisplayName("Date1")]
            public DateTime? Date1 { get { return Fields.Date1[this]; } set { Fields.Date1[this] = value; } }
            public partial class RowFields { public DateTimeField Date1; }
            #endregion Date1
                
            #region Date2
            [DisplayName("Date2")]
            public DateTime? Date2 { get { return Fields.Date2[this]; } set { Fields.Date2[this] = value; } }
            public partial class RowFields { public DateTimeField Date2; }
            #endregion Date2
                
            #region Date3
            [DisplayName("Date3")]
            public DateTime? Date3 { get { return Fields.Date3[this]; } set { Fields.Date3[this] = value; } }
            public partial class RowFields { public DateTimeField Date3; }
            #endregion Date3
                
            #region Date4
            [DisplayName("Date4")]
            public DateTime? Date4 { get { return Fields.Date4[this]; } set { Fields.Date4[this] = value; } }
            public partial class RowFields { public DateTimeField Date4; }
            #endregion Date4
                
            #region Date5
            [DisplayName("Date5")]
            public DateTime? Date5 { get { return Fields.Date5[this]; } set { Fields.Date5[this] = value; } }
            public partial class RowFields { public DateTimeField Date5; }
            #endregion Date5
                
            #region Category1
            [DisplayName("Category1"), ForeignKey("[adr].[Category1]", "Category1ID"), LeftJoin("jCategory1"), TextualField("Category1Category1Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category1Row), InplaceAdd = true)]
            public Int32? Category1 { get { return Fields.Category1[this]; } set { Fields.Category1[this] = value; } }
            public partial class RowFields { public Int32Field Category1; }
            #endregion Category1
                
            #region Category2
            [DisplayName("Category2"), ForeignKey("[adr].[Category2]", "Category2ID"), LeftJoin("jCategory2"), TextualField("Category2Category2Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category2Row), InplaceAdd = true)]
            public Int32? Category2 { get { return Fields.Category2[this]; } set { Fields.Category2[this] = value; } }
            public partial class RowFields { public Int32Field Category2; }
            #endregion Category2
                
            #region Category3
            [DisplayName("Category3"), ForeignKey("[adr].[Category3]", "Category3ID"), LeftJoin("jCategory3"), TextualField("Category3Category3Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category3Row), InplaceAdd = true)]
            public Int32? Category3 { get { return Fields.Category3[this]; } set { Fields.Category3[this] = value; } }
            public partial class RowFields { public Int32Field Category3; }
            #endregion Category3
                
            #region Category4
            [DisplayName("Category4"), ForeignKey("[adr].[Category4]", "Category4ID"), LeftJoin("jCategory4"), TextualField("Category4Category4Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category4Row), InplaceAdd = true)]
            public Int32? Category4 { get { return Fields.Category4[this]; } set { Fields.Category4[this] = value; } }
            public partial class RowFields { public Int32Field Category4; }
            #endregion Category4
                
            #region Category5
            [DisplayName("Category5"), ForeignKey("[adr].[Category5]", "Category5ID"), LeftJoin("jCategory5"), TextualField("Category5Category5Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category5Row), InplaceAdd = true)]
            public Int32? Category5 { get { return Fields.Category5[this]; } set { Fields.Category5[this] = value; } }
            public partial class RowFields { public Int32Field Category5; }
            #endregion Category5
                
            #region Category6
            [DisplayName("Category6"), ForeignKey("[adr].[Category6]", "Category6ID"), LeftJoin("jCategory6"), TextualField("Category6Category6Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category6Row), InplaceAdd = true)]
            public Int32? Category6 { get { return Fields.Category6[this]; } set { Fields.Category6[this] = value; } }
            public partial class RowFields { public Int32Field Category6; }
            #endregion Category6
                
            #region Category7
            [DisplayName("Category7"), ForeignKey("[adr].[Category7]", "Category7ID"), LeftJoin("jCategory7"), TextualField("Category7Category7Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category7Row), InplaceAdd = true)]
            public Int32? Category7 { get { return Fields.Category7[this]; } set { Fields.Category7[this] = value; } }
            public partial class RowFields { public Int32Field Category7; }
            #endregion Category7
                
            #region Category8
            [DisplayName("Category8"), ForeignKey("[adr].[Category8]", "Category8ID"), LeftJoin("jCategory8"), TextualField("Category8Category8Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category8Row), InplaceAdd = true)]
            public Int32? Category8 { get { return Fields.Category8[this]; } set { Fields.Category8[this] = value; } }
            public partial class RowFields { public Int32Field Category8; }
            #endregion Category8
                
            #region Category9
            [DisplayName("Category9"), ForeignKey("[adr].[Category9]", "Category9ID"), LeftJoin("jCategory9"), TextualField("Category9Category9Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category9Row), InplaceAdd = true)]
            public Int32? Category9 { get { return Fields.Category9[this]; } set { Fields.Category9[this] = value; } }
            public partial class RowFields { public Int32Field Category9; }
            #endregion Category9
                
            #region Category10
            [DisplayName("Category10"), ForeignKey("[adr].[Category10]", "Category10ID"), LeftJoin("jCategory10"), TextualField("Category10Category10Name")]
            [LookupEditor(typeof(AddressBookDB.Entities.Category10Row), InplaceAdd = true)]
            public Int32? Category10 { get { return Fields.Category10[this]; } set { Fields.Category10[this] = value; } }
            public partial class RowFields { public Int32Field Category10; }
            #endregion Category10
                
            #region Creation Date
            [DisplayName("Creation Date"), Visible(false)]
            public DateTime? CreationDate { get { return Fields.CreationDate[this]; } set { Fields.CreationDate[this] = value; } }
            public partial class RowFields { public DateTimeField CreationDate; }
            #endregion CreationDate
                
            #region Last Update Date
            [DisplayName("Last Update Date"), Visible(false)]
            public DateTime? LastUpdateDate { get { return Fields.LastUpdateDate[this]; } set { Fields.LastUpdateDate[this] = value; } }
            public partial class RowFields { public DateTimeField LastUpdateDate; }
            #endregion LastUpdateDate
                
            #region Last Update User
            [DisplayName("Last Update User"), Size(100), Visible(false)]
            public String LastUpdateUser { get { return Fields.LastUpdateUser[this]; } set { Fields.LastUpdateUser[this] = value; } }
            public partial class RowFields { public StringField LastUpdateUser; }
            #endregion LastUpdateUser
                
            #region Last Update Prg
            [DisplayName("Last Update Prg"), Size(100), Visible(false)]
            public String LastUpdatePrg { get { return Fields.LastUpdatePrg[this]; } set { Fields.LastUpdatePrg[this] = value; } }
            public partial class RowFields { public StringField LastUpdatePrg; }
            #endregion LastUpdatePrg
        

    #region Foreign Fields
            
                [DisplayName("Address Type Address Type Name"), Expression("jAddressType.[AddressTypeName]")]
                public String AddressTypeAddressTypeName { get { return Fields.AddressTypeAddressTypeName[this]; } set { Fields.AddressTypeAddressTypeName[this] = value; } }
                public partial class RowFields { public StringField AddressTypeAddressTypeName; }

                        
                [DisplayName("Constituency Constituency Name"), Expression("jConstituency.[ConstituencyName]")]
                public String ConstituencyConstituencyName { get { return Fields.ConstituencyConstituencyName[this]; } set { Fields.ConstituencyConstituencyName[this] = value; } }
                public partial class RowFields { public StringField ConstituencyConstituencyName; }

                        
                [DisplayName("Voting Center Voting Center Name"), Expression("jVotingCenter.[VotingCenterName]")]
                public String VotingCenterVotingCenterName { get { return Fields.VotingCenterVotingCenterName[this]; } set { Fields.VotingCenterVotingCenterName[this] = value; } }
                public partial class RowFields { public StringField VotingCenterVotingCenterName; }

                        
                [DisplayName("Category1 Category1 Name"), Expression("jCategory1.[Category1Name]")]
                public String Category1Category1Name { get { return Fields.Category1Category1Name[this]; } set { Fields.Category1Category1Name[this] = value; } }
                public partial class RowFields { public StringField Category1Category1Name; }

                        
                [DisplayName("Category2 Category2 Name"), Expression("jCategory2.[Category2Name]")]
                public String Category2Category2Name { get { return Fields.Category2Category2Name[this]; } set { Fields.Category2Category2Name[this] = value; } }
                public partial class RowFields { public StringField Category2Category2Name; }

                        
                [DisplayName("Category3 Category3 Name"), Expression("jCategory3.[Category3Name]")]
                public String Category3Category3Name { get { return Fields.Category3Category3Name[this]; } set { Fields.Category3Category3Name[this] = value; } }
                public partial class RowFields { public StringField Category3Category3Name; }

                        
                [DisplayName("Category4 Category4 Name"), Expression("jCategory4.[Category4Name]")]
                public String Category4Category4Name { get { return Fields.Category4Category4Name[this]; } set { Fields.Category4Category4Name[this] = value; } }
                public partial class RowFields { public StringField Category4Category4Name; }

                        
                [DisplayName("Category5 Category5 Name"), Expression("jCategory5.[Category5Name]")]
                public String Category5Category5Name { get { return Fields.Category5Category5Name[this]; } set { Fields.Category5Category5Name[this] = value; } }
                public partial class RowFields { public StringField Category5Category5Name; }

                        
                [DisplayName("Category6 Category6 Name"), Expression("jCategory6.[Category6Name]")]
                public String Category6Category6Name { get { return Fields.Category6Category6Name[this]; } set { Fields.Category6Category6Name[this] = value; } }
                public partial class RowFields { public StringField Category6Category6Name; }

                        
                [DisplayName("Category7 Category7 Name"), Expression("jCategory7.[Category7Name]")]
                public String Category7Category7Name { get { return Fields.Category7Category7Name[this]; } set { Fields.Category7Category7Name[this] = value; } }
                public partial class RowFields { public StringField Category7Category7Name; }

                        
                [DisplayName("Category8 Category8 Name"), Expression("jCategory8.[Category8Name]")]
                public String Category8Category8Name { get { return Fields.Category8Category8Name[this]; } set { Fields.Category8Category8Name[this] = value; } }
                public partial class RowFields { public StringField Category8Category8Name; }

                        
                [DisplayName("Category9 Category9 Name"), Expression("jCategory9.[Category9Name]")]
                public String Category9Category9Name { get { return Fields.Category9Category9Name[this]; } set { Fields.Category9Category9Name[this] = value; } }
                public partial class RowFields { public StringField Category9Category9Name; }

                        
                [DisplayName("Category10 Category10 Name"), Expression("jCategory10.[Category10Name]")]
                public String Category10Category10Name { get { return Fields.Category10Category10Name[this]; } set { Fields.Category10Category10Name[this] = value; } }
                public partial class RowFields { public StringField Category10Category10Name; }

            
    #endregion Foreign Fields

    #region Id and Name fields
    IIdField IIdRow.IdField
    {
    get { return Fields.AddressId; }
    }
        
            StringField INameRow.NameField
            {
            get { return Fields.FullName; }
            }
            #endregion Id and Name fields

    #region Constructor
    public AddressBookRow()
    : base(Fields)
    {
    }
    #endregion Constructor

    #region RowFields
    public static readonly RowFields Fields = new RowFields().Init();

    public partial class RowFields : RowFieldsBase
    {
    public RowFields()
    : base("[adr].[AddressBook]")
    {
    LocalTextPrefix = "AddressBookDB.AddressBook";
    }
    }
    #endregion RowFields
    }
    }
`

//AddressBookColumns.cs
namespace AddressDB.AddressBookDB.Columns
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using System;
    using System.ComponentModel;
    using System.Collections.Generic;
    using System.IO;

    [ColumnsScript("AddressBookDB.AddressBook")]
    [BasedOnRow(typeof(Entities.AddressBookRow))]
    public class AddressBookColumns
    {
        [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
        public Int32 AddressId { get; set; }
        public Int32 AddressType { get; set; }
        [EditLink]
        public String FullName { get; set; }

        //[Hidden]
        public String LastName { get; set; }
        //[Hidden]
        public String FirstName { get; set; }
        [EditLink]
        public Entities.Gender Gender { get; set; }
        public String FatherName { get; set; }
        public String MotherName { get; set; }
        public String Address { get; set; }
        public String City { get; set; }
        public String PostCode { get; set; }
        public String Profession { get; set; }
        public DateTime BirthDate { get; set; }
        public Int32 Constituency { get; set; }
        public Int32 VotingCenter { get; set; }
        public String PollingStation { get; set; }
        //public String Description { get; set; }
        //public String Notes { get; set; }
        //public String Photo { get; set; }
        //public Int32 Number1 { get; set; }
        //public Int32 Number2 { get; set; }
        //public Int32 Number3 { get; set; }
        //public Int32 Number4 { get; set; }
        //public Int32 Number5 { get; set; }
        //public DateTime Date1 { get; set; }
        //public DateTime Date2 { get; set; }
        //public DateTime Date3 { get; set; }
        //public DateTime Date4 { get; set; }
        //public DateTime Date5 { get; set; }
        //public Int32 Category1 { get; set; }
        //public Int32 Category2 { get; set; }
        //public Int32 Category3 { get; set; }
        //public Int32 Category4 { get; set; }
        //public Int32 Category5 { get; set; }
        //public Int32 Category6 { get; set; }
        //public Int32 Category7 { get; set; }
        //public Int32 Category8 { get; set; }
        //public Int32 Category9 { get; set; }
        //public Int32 Category10 { get; set; }
        //public DateTime CreationDate { get; set; }
        //public DateTime LastUpdateDate { get; set; }
        //public String LastUpdateUser { get; set; }
        //public String LastUpdatePrg { get; set; }
    }
}

//AddressBookGrid.ts
namespace AddressDB.AddressBookDB {
    
    @Serenity.Decorators.registerClass()
    export class AddressBookGrid extends Serenity.EntityGrid<AddressBookRow, any> {
        protected getColumnsKey() { return 'AddressBookDB.AddressBook'; }
        protected getDialogType() { return AddressBookDialog; }
        protected getIdProperty() { return AddressBookRow.idProperty; }
        protected getLocalTextPrefix() { return AddressBookRow.localTextPrefix; }
        protected getService() { return AddressBookService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}

**Comment by volkanceylan** [comment]
C# is a case sensitive language, also javascript. So write FullName everywhere. You wrote FullName in row while Fullname in property while FullName in field

**Comment by dkontod** [comment]
@volkanceylan Thank you very much. You have probably saved my project. I apologize for all the inconvenience I've caused. I knew of course that c# is case sensitive and I tried to fix all occurrences of FullName. But the time was late and I missed one. There was another one though that had a 'N' in Greek capital letter and the Find-Replace did not catch it. But night's works, day's jokes as we say here. 
Where can I ask for professional help (cooperaration) upon another project with serene?
